### PR TITLE
perf(softmax): strip trivial reshapes to allow `fused_softmax` on size-1 axes

### LINKED ISSUE #200: `fuse_softmax` misses softmax when reduced shape has unit dims
`fuse_softmax.cc` fails to match `jax.nn.softmax` for shapes with size-1 dims like `(1, N)` with axis=-1.

For this shape, StableHLO simplification represents the `-inf` guard as:

```mlir
%1 = stablehlo.reshape %cst_0 : (tensor<f32>) -> tensor<1xf32>
%2 = stablehlo.maximum %1, %0 : tensor<1xf32>
```
instead of the broadcast form that the matcher expects: `maximum(broadcast(-inf), reduce_max)`

As a result, `stripNegInfMaximum()` does not recognize `reshape(-inf)` as the identity operand, leaves the maximum in place, and the later `matchSingleAxisReduce<stablehlo::MaxOp>` check fails. The optimized IR keeps the full reduce/sub/exp/reduce/divide softmax chain instead of producing `mps.softmax`.

Repro:

```python
x = jax.random.normal(jax.random.PRNGKey(0), (1, 2**26,))
f = jax.jit(jax.nn.softmax)

_ = f(x).block_until_ready()  # Warmup

for _ in range(10):
  t0 = time.monotonic()
  y = f(x).block_until_ready()
  t1 = time.monotonic()
  print(f"Time: {(t1 - t0) * 1e3:.6f} ms")

# ~15ms for (2**26,) identifies as `mps.softmax`
# ~23ms for (1, 2**26) falls back to trivial chain.
```

Fusion tests do not catch this because dummy tensors do not contain size-1 dim tensors. 

The fix is a one-liner canonicalization of removing trivial reshapes within `stripNegInfMaximum`. I can raise a PR if this looks OK.

### PR DESCRIPTION
**Pull request by MasterSkepticista**
Fixes #200.

Earlier, the pattern matcher could identify:
```
maximum(broadcast(-inf), reduce_max)
          |
          recognized as -inf
```
for all tensor shapes without size-1 axes. 

StableHLO spec [says](https://openxla.org/stablehlo/generated/stablehlo_optimization_passes#-stablehlo-aggressive-simplification) that a `broadcast_in_dim` must become a `reshape` if number of elements is the same (which is true when `-inf` is "broadcasted" to a new tensor with size-1 axes).

```
maximum(reshape(-inf), reduce_max)
          |
          not recognized as -inf => no fusion
```

This PR strips trivial reshapes on the operand to allow pattern matching.

Tests updated for the same.

### CODE REVIEW

**Comment by tillahoffmann** [comment]
Nice, thank you!
